pkg/ccl/streamingccl/streamingest/stream_ingestion_processor.go

```diff
@@ -643,37 +731,127 @@ func (sip *streamIngestionProcessor) bufferCheckpoint(event partitionEvent) erro
 	return nil
 }
 
+// Write a batch of MVCC range keys into the SST batcher, and returns
+// the current size of all buffered range keys.
+func (r *rangeKeyBatcher) write(rangeKeys mvccRangeKeyValues) (int, error) {
+	if len(rangeKeys) == 0 {
+		return 0, nil
+	}
+	// Lazily create the SST writer when we really need it.
+	if r.rangeKeySSTWriter == nil {
+		r.rangeKeySSTWriter = r.rangeKeySSTWriterMaker()
+	}
+
+	for _, rangeKeyVal := range rangeKeys {
```

Comment:
Are there ordering constraints on how we write range keys into the SST? That is, do we need sort this like we sort KV keys?

Reply:
Yes, they must be written in key order. Multiple versions of the same range key can be written in any order, as long as the key order is maintained.